sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.PartitionProcessor.cs

```diff
@@ -218,13 +217,29 @@ public async Task ProcessEventsAsync(EventProcessorHostPartition context, IEnume
                 // so that is the responsibility of the user's function currently. E.g.
                 // the function should have try/catch handling around all event processing
                 // code, and capture/log/persist failed events, since they won't be retried.
-                if (eventToCheckpoint != null)
+                if (eventToCheckpoint == null)
                 {
-                    await CheckpointAsync(eventToCheckpoint, context).ConfigureAwait(false);
+                    return;
                 }
+
+                // If the partition processing token is canceled but the overall processor token is NOT canceled, this means we likely lost ownership
+                // of the partition and should not checkpoint. This is best effort and not thread-safe.
+                if (partitionCancellationToken.IsCancellationRequested && !_cts.IsCancellationRequested)
```

Comment:
My understanding is that the partitionCancellationToken is a linked token consisting of the overall processor token and a token that is signaled when partition ownership is lost. The _cts token is just the overall processor token.

Reply:
After writing this out, I realized that this isn't correct. The _cts token is also cancelled for ownership loss.